File: Code/new_fitervari_backend/src/main/java/at/htl/control/SetHistoryRepository.java
package at.htl.control;

import at.htl.model.SetHistory;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class SetHistoryRepository implements PanacheRepository<SetHistory> {

    public SetHistory save(SetHistory entity) {
        return getEntityManager().merge(entity);
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/SetHistoryRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.SetHistory
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class SetHistoryRepository : PanacheRepository<SetHistory> {
    @Inject
    lateinit var em: EntityManager

    fun save(setHistory: SetHistory): SetHistory {
        return em.merge(setHistory)
    }
}